src/vs/workbench/contrib/codeEditor/browser/saveParticipants.ts

```diff
@@ -319,6 +318,28 @@ class CodeActionOnSaveParticipant implements ITextFileSaveParticipant {
 		await this.applyOnSaveActions(textEditorModel, codeActionsOnSave, excludedActions, progress, token);
 	}
 
+	private createCodeActionsOnSave(settingItems: string[]): CodeActionKind[] {
+		const actionSeeds = new Set<string>();
+
+		// Remove subsets
+		const len = settingItems.length;
```

Comment:
Instead of nested for loops, can you use `array.filter`? That should simplify the logic